### PR TITLE
JENKINS-56109 Compatible JavaScript changes for tables to divs

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-56109](https://issues.jenkins-ci.org/browse/JENKINS-56109).
and https://github.com/jenkinsci/jenkins/pull/3895

I've extracted all of the compatible JavaScript changes from tables to divs

The non compatible changes are all related to elements moving to different locations of components that are managed in core, i.e. validation error areas and the hetero-list config page.

This make it so:
1. Easier to review
2. We can start landing smaller PRs in different weeklies to catch any issues
3. If we do have to fallback because of a critical issue it would make tables to divs easier to revert

I've clicked around freestyle, pipeline, tool configuration, manage jenkins, and checked matrix-auth which was one of the plugins that has issues on the main PR.

### Proposed changelog entries

Unneeded? There's a chance of regression but the aim of this is there to be no user facing changes in this PR

* Entry 1: JENKINS-56109: Internal: JavaScript refactoring in preparation for changing form markup from tables to divs
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by fqueiruga** [comment]
Thank you for creating this, it will really help. I'm a bit concern that the changes in `findFollowingTR` may cause issues, we'll need to test those uses thoroughly.

**Comment by oleg-nenashev** [comment]
I think an internal changelog would be nice. Looking into the code

**Review by jtnord — APPROVED** [review]
looks ok at a quick glance

**Review by jsoref — APPROVED** [review]
Sure

**Comment by timja** [comment]
> git-parameter-plugin

I can't see any issues with the plugins I have installed.

I just checked the above plugin and all seems fine:

image

**Comment by fqueiruga** [comment]
The problem with git-parameter-plugin came when populating the `.validation-error-area` box on the "Build with parameters view. One of the fixes I added 2 months ago was for this.

Captura de pantalla 2020-06-26 a las 9 03 25

**Comment by timja** [comment]
ah ok

**Review by fqueiruga — APPROVED** [review]
Tested it locally and haven't been able to make it crash

**Comment by timja** [comment]
This PR is ready for merge, we'll merge it after 24 hours if there's no negative feedback

**Comment by zbynek** [comment]
This introduced an exception when you focus/blur the search field, since the scope of `defaultValue` has changed
https://github.com/jenkinsci/jenkins/commit/df3b7ce9d48c74d7bc2c34d1d51ac23ca9747412#diff-da911448bbb85d374f604da9c1e00136L665

**Comment by timja** [comment]
@zbynek https://github.com/jenkinsci/jenkins/pull/4876

**Comment by Wadeck** [comment]
Regression discovered for the help button in dropdownList tag: https://issues.jenkins-ci.org/browse/JENKINS-63910 / https://github.com/jenkinsci/jenkins/pull/4986

**Comment by mmh891113** [comment]
@timja
Hi tim,  What do you think of this problem JENKINS-63931, https://issues.jenkins-ci.org/projects/JENKINS/issues/JENKINS-63931?filter=allopenissues
this issue should install ( Extended Choice Parameter Plug-In)

**Comment by timja** [comment]
I assume it's something to do with sections being in radio blocks.

Maybe try remove the section?

**Comment by mmh891113** [comment]
@vimil 
the section tag just to show the title Information. Maybe try remove the section  or replace `<f:section` to other tag???
hi vimil what do you think of this problem JENKINS-63931,

**Comment by timja** [comment]
best to discuss this in the jira issue.

File: war/src/main/webapp/scripts/hudson-behavior.js [hunk 1]
```diff
@@ -1265,17 +1329,51 @@ var jenkinsRules = {
...
+     * It will also move the help button.
+     *
+     * @param {HTMLLabelElement} label
+     */
+    Behaviour.specify('label.js-checkbox-label-empty', 'form-fallbacks', 1000, function(label) {
+        var labelParent = label.parentElement;
+        if (!labelParent.classList.contains('setting-main')) return;
+
+        function findSettingName(formGroup) {
+            for (var i=0; i<formGroup.childNodes.length; i++) {
```
Comment (basil) [inline war/src/main/webapp/scripts/hudson-behavior.js:1351]: Caused [JENKINS-65511](https://issues.jenkins.io/browse/JENKINS-65511) as demonstrated in #6739.

**Pull request merged**